ls1intum/Artemis
# Enhancement/Code Editor Instructor View: UI improvements

### Checklist

### Motivation and Context



Improve the instructor/ta screens for programming-exercises so that they are easier to use.

### Description


**Code-Editor:**
- Feature: Add resizing option for build output.
- Enhancement: Remove checkmark for indicating that file loading is done in ace editor.
- Styling: Make action buttons in file-browser lighter.
- Styling: Make vertical/horizontal resizing work as expected.
- Styling: Structure editor styles into component scss modules (before that there was a global file).

**Code-Editor-Instructor-View:**
- Enhancement: Don't reload the screen on first repo change (There was a route for not selecting a participation, this caused a visible reload of the page when a repository is selected the first time).
- Bugfix: Don't add the participation to the participation websocket cache (This could cause issues on other pages as the course-exercise-detail).

**Editable-Instructions:**
- Feature: Move the save problem statement button into the editable programming exercise component. This way also in the update/dialog programming exercise screen the problem statement can be saved invidividually.
- Feature: Move status icons into bottom area of editable instructions (similiar to file-browser).
- Enhancement: Hide status & save button if programming exercise is being created in update screen
- Bugfix: Also hide the update oracle structure tooltip, when the button is hidden.
- Styling: Fix command button styling in markdown editor (weird linebreaks on smaller devices).
- Styling: Increase default size of editable instructions
- Styling: Structure programming-exercise instruction styles into component scss modules (before that there was a global file).

**Markdown-Editor:**
- Feature: Add fullscreen mode to markdown editor
- Styling: Structure markdown-editor instruction styles into component a scss module. This way the markdown-editor adjusts its size now to the parent component.

:warning: Known styling issues in the code-editor :warning: 
1) In the instructor code editor, the height of the main area jumps a little, when switching from editor to preview in the instructions. This issue is very hard to pin down, I wouldn't invest more time into it personally.
2) In the code editor on very small screen when adjusting the width of the instructions it will overflow the main container to the right. This issue is also hard to solve (part of the max min width calculation needs to be done in JS because of InteractJs) and the best compromise to a resizable layout. I don't think it's a major issue because students can still just collapse the instructions.

### Steps for Testing


Every Feature/Enhancement/Bugfix has its own testing steps (see description above).
Most of the changes concern styling.

All components that are concerned by the changes:
- Code-Editor Student
- Code-Editor Instructor
- Programming-Exercise update/dialog/detail
- Markdown-Editor (and all components that use the markdown editor, only styling wise however)

### Screenshots



Code-Editor-Instructor before:

![image](https://user-images.githubusercontent.com/28230611/60431749-cc1e9480-9c00-11e9-8aa1-e5ecbd761301.png)

Code-Editor-Instructor after:

![image](https://user-images.githubusercontent.com/28230611/60431683-a2656d80-9c00-11e9-8a2f-e920ed9885c0.png)

Markdown-Fullscreen mode:

![image](https://user-images.githubusercontent.com/28230611/60431848-11db5d00-9c01-11e9-96cc-da82cd6ab0c7.png)
